src/test/java/com/redhat/darcy/webdriver/TakeScreenshotTest.java

```diff
@@ -1,49 +1,95 @@
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
+import com.redhat.darcy.ui.DarcyException;
 import com.redhat.darcy.ui.FindableNotPresentException;
 import com.redhat.darcy.web.api.Browser;
 import com.redhat.darcy.webdriver.internal.TargetedWebDriver;
 import com.redhat.darcy.webdriver.testing.doubles.StubWebDriverElementContext;
 import com.redhat.darcy.webdriver.testing.doubles.StubWebDriverParentContext;
 
-import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.mockito.Mock;
-import org.mockito.runners.MockitoJUnitRunner;
+import org.junit.runners.JUnit4;
+import org.mockito.InOrder;
 import org.openqa.selenium.NoSuchWindowException;
 import org.openqa.selenium.OutputType;
 
-@RunWith(MockitoJUnitRunner.class)
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+@RunWith(JUnit4.class)
 public class TakeScreenshotTest {
-    @Mock
-    TargetedWebDriver mockedDriver;
-
-    Browser browser;
-
-    @Before
-    public void setupBrowser() {
-        if (browser == null) {
-            browser = new WebDriverBrowser(mockedDriver,
-                    new StubWebDriverParentContext(),
-                    new StubWebDriverElementContext());
-        }
+    @Test
+    public void shouldTakeScreenshotAndWriteToOutputStream() throws IOException {
+        TargetedWebDriver mockedDriver = mock(TargetedWebDriver.class);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        Browser browser = new WebDriverBrowser(mockedDriver,
+                new StubWebDriverParentContext(),
+                new StubWebDriverElementContext());
+
+        byte[] data = new byte[] { 1, 2, 3 };
+
+        when(mockedDriver.getScreenshotAs(OutputType.BYTES))
+                .thenReturn(data);
+
+        browser.takeScreenshot(baos);
+
+        assertThat(baos.toByteArray(), equalTo(data));
     }
 
     @Test
-    public void shouldTakeScreenshot() {
-        browser.takeScreenshot();
-        verify(mockedDriver).getScreenshotAs(OutputType.FILE);
+    public void shouldTakeScreenshotAndWriteToOutputStreamThenFlushAndCloseInOrder() throws IOException {
+        TargetedWebDriver mockedDriver = mock(TargetedWebDriver.class);
+        OutputStream outputStream = mock(OutputStream.class);
+        Browser browser = new WebDriverBrowser(mockedDriver,
+                new StubWebDriverParentContext(),
+                new StubWebDriverElementContext());
+
+        byte[] data = new byte[] { 1, 2, 3 };
+
+        when(mockedDriver.getScreenshotAs(OutputType.BYTES))
+                .thenReturn(data);
+        InOrder inOrder = inOrder(outputStream);
+
+        browser.takeScreenshot(outputStream);
+
+        inOrder.verify(outputStream).write(data);
```

Comment:
If you're going to verify the data is passed in here, then the test that uses a real `OutputStream` is superfluous. If you really want separate tests then I would use `any(Byte[].class)` here.

Probably best of all would be to not use a runtime-mock and instead use an implementation of `OutputStream` that just requires it is used correctly or it doesn't work right, but it's not a huge deal. (One of the benefits of doing that would be we would avoid having to care which `write` method is used, for instance.)